arch/wasm/bits/syscall.h.in

```diff
@@ -0,0 +1,325 @@
+// The list of syscalls is taken from x32, which contains a sensible list of
+// "modern" syscalls.
+
+#define SYSCALL_ACTION(x) long __syscall_##x(long arg1, ...);
```

Comment:
Some syscalls take a variable number of arguments (like SYS_open, and SYS_mremap).

I thought it would be cunning to therefore give syscalls a varargs signature. (NB. The way static linking for syscalls has to work is that all syscalls must have the same signature. See how support for `syscall()` and `pthread_cancel()` was implemented by cunningly using each syscall's function pointer as its syscall number! Hat-tip to John Starks at Microsoft.)

Making them varargs is a silly idea, I should just make all take six normal arguments, and pass zeros for the ones that don't matter.